File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsAlternative/QuoteContractAlternativePPMVisit/QuoteContractAlternativePPMVisitValidator.vb
Namespace Entity

    Namespace QuoteContractAlternativePPMVisits

        Public Class QuoteContractAlternativePPMVisitValidator
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oQuoteContractPPMVisit As QuoteContractAlternativePPMVisit)

                'make sure that contact object is valid
                If oQuoteContractPPMVisit.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oQuoteContractPPMVisit.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If



                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsAlternative/QuoteContractAlternativePPMVisit/QuoteContractAlternativePPMVisitValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace QuoteContractAlternativePPMVisits
    {
        // make sure that contact object is valid
        public class QuoteContractAlternativePPMVisitValidator : BaseValidator
        {
            public void Validate(QuoteContractAlternativePPMVisit oQuoteContractPPMVisit)
            {
                // make sure that contact object is valid
                if (oQuoteContractPPMVisit.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oQuoteContractPPMVisit.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}